clang/lib/CodeGen/Targets/AArch64.cpp

```diff
@@ -353,6 +427,17 @@ AArch64ABIInfo::classifyArgumentType(QualType Ty, bool IsVariadic,
         nullptr, true, Align);
   }
 
+  // In AAPCS named arguments of a Pure Scalable Type are passed expanded in
+  // registers, or indirectly if there are not enough registers.
+  if (Kind == AArch64ABIKind::AAPCS && !IsVariadic) {
```

Comment:
What spec rule excludes variadic functions?

Reply:
That's Rule C.7, that wants the argument to be named.

> If the argument is a Pure Scalable Type that consists of NV Scalable Vector Types and
> NP Scalable Predicate Types, **if the argument is named**, if NSRN+NV ≤ 8,
> and if NPRN+NP ≤ 4, then the Scalable Vector Types are

Reply:
I'm pretty sure "IsVariadic" is whether the function type is variadic, not whether the argument is named.

Reply:
Indeed! Fixed.